frontend/dockerfile/instructions/parse.go

```diff
@@ -362,13 +362,14 @@ func parseFrom(req parseRequest) (*Stage, error) {
 	}, nil
 }
 
-func parseBuildStageName(args []string) (string, error) {
-	stageName := ""
+var validStageName = regexp.MustCompile("^[a-z][a-z0-9-_.]*$")
+
+func parseBuildStageName(args []string) (stageName string, err error) {
 	switch {
 	case len(args) == 3 && strings.EqualFold(args[1], "as"):
 		stageName = strings.ToLower(args[2])
-		if ok, _ := regexp.MatchString("^[a-z][a-z0-9-_\\.]*$", stageName); !ok {
-			return "", errors.Errorf("invalid name for build stage: %q, name can't start with a number or contain symbols", args[2])
+		if !validStageName.MatchString(stageName) {
+			return "", errors.Errorf("invalid name for build stage: %q, name can't start with a number or contain symbols", stageName)
```

Comment:
nit: this will include the lowercased version in the error message - is that what we want?

Reply:
Oh! Good catch; I initially made the regex itself case-insensitive (adding `(?i)`) but I saw that it introduces more allocations (or memory use) than the existing `strings.ToLower()` so I changed that back, but forgot to put back this value here.

Let me update that.